scripts/tutorial-uploader/lib/api.test.ts

```diff
@@ -0,0 +1,154 @@
+        fields: ["*", { translations: ["*"] }],
+      }),
+    );
+    expect(retrievedTutorial).toMatchObject({
+      slug: MOCK_TUTORIAL.slug,
+      status: MOCK_TUTORIAL.status,
+      reading_time: MOCK_TUTORIAL.reading_time,
+      catalog_featured: MOCK_TUTORIAL.catalog_featured,
+      category: await api.getId(
+        "tutorials_categories",
+        "name",
+        MOCK_TUTORIAL.category,
+      ),
+      topics: [],
+      allowed_email_domains: ["ibm.com", "hotmail.co.uk"],
+      required_instance_access: ["ibm-quantum/group/project"],
+      sort: null,
+      translations: [
+        {
+          title: MOCK_TUTORIAL.title,
+          short_description: MOCK_TUTORIAL.short_description,
+          content: "Here's some basic content.\n",
+          languages_code: "en-US",
+        },
+      ],
+    });
+  });
+
+  test("update existing tutorial", async () => {
+    // Upload tutorial
+    await api.upsertTutorial(MOCK_TUTORIAL);
+    const tutorialId = await api.getTutorialIdBySlug(MOCK_TUTORIAL.slug);
+
+    // Mutate tutorial data and re-upload
+    const modifiedTutorial = {
+      ...MOCK_TUTORIAL,
+      title: "A new tutorial title",
+      short_description: "A modified short description",
+      status: "draft",
+      category: "How-to",
+      reading_time: 33,
+      topics: ["Scheduling", "Transpilation"],
+      catalog_featured: false,
+    };
+    await api.upsertTutorial(modifiedTutorial);
+
+    // Retrieve and check
+    const retrievedTutorial = await api.client.request(
+      readItem("tutorials", tutorialId as string, {
+        fields: ["*", { topics: ["*"] }],
+      }),
+    );
+    const topicIds = (await Promise.all(
+      modifiedTutorial.topics.map((name) =>
+        api.getId("tutorials_topics", "name", name),
+      ),
+    )) as string[];
+    expect(retrievedTutorial).toMatchObject({
+      slug: modifiedTutorial.slug,
+      status: modifiedTutorial.status,
+      reading_time: modifiedTutorial.reading_time,
+      catalog_featured: modifiedTutorial.catalog_featured,
+      category: await api.getId(
+        "tutorials_categories",
+        "name",
+        modifiedTutorial.category,
+      ),
+      topics: topicIds.map((id) => {
+        return { tutorials_topics_id: id };
+      }),
+      allowed_email_domains: ["ibm.com", "hotmail.co.uk"],
+      required_instance_access: ["ibm-quantum/group/project"],
+      sort: null,
+    });
+
+    const retrievedTranslation = (
+      await api.client.request(
+        readItem("tutorials", tutorialId as string, {
+          fields: [{ translations: ["*"] }],
+        }),
+      )
+    ).translations;
+    expect(retrievedTranslation).toMatchObject([
+      {
+        title: modifiedTutorial.title,
+        short_description: modifiedTutorial.short_description,
+        content: "Here's some basic content.\n",
+        languages_code: "en-US",
+      },
+    ]);
+  });
+
+  beforeAll(async () => {
```

Comment:
Nit that the convention is for this and `afterEach` to be near the top of the describe block